packages/components/src/common/dimensions.ts

```diff
@@ -16,3 +16,69 @@ export const unmeasuredDimensions: ElementDimensions = {
 export interface DimensionsById {
     [id: string]: ElementDimensions;
 }
+
+export const observeElementDimensions = (element: HTMLElement, callback: (dimensions: ElementDimensions) => void) => {
+    const observer = new ResizeObserver(() => callback(element.getBoundingClientRect()));
+
+    if (element !== null) return;
+
+    return () => observer.disconnect();
+};
+
+export const observeWindowDimensions = (callback: (dimensions: ElementDimensions) => void) => {
+    const onResize = () =>
+        callback({
+            width: window.innerWidth,
+            height: window.innerHeight,
+        });
+
+    onResize();
+
+    window.addEventListener('resize', onResize, {
+        capture: false,
+        passive: true,
+    });
+
+    return () => window.removeEventListener('resize', onResize);
+};
+
+export const getSizeFromDimensions = (dimensions: ElementDimensions, sizeAsHeight: boolean): number => {
+    return (sizeAsHeight ? dimensions.height : dimensions.width) ?? 0;
+};
+
+export const getDimensionsFromRect = (rect?: DOMRect): ElementDimensions => {
+    return rect
+        ? {
+              width: rect.width,
+              height: rect.height,
+          }
+        : unmeasuredDimensions;
+};
+
+export const getElementDimensions = (element: HTMLElement | null | undefined): ElementDimensions => {
```

Comment:
so if I pass null or undefined , I get window dimensions ?
it is not expected from function name

Reply:
Discussed, will be executed later in stylable-components